### DISCUSSION TITLE
Gmail classifies Bitwarden RS emails as spam

### DISCUSSION THREAD

**Discussion by FractalBoy** [dani-garcia/vaultwarden#1354] [Q&A]
I've configured SPF, DKIM, DMARC, and set up a PTR record and I've managed to get my mail-tester.org score up to 9.5. However, there are things that mail-tester is finding that I cannot address.

This is from a test email, so it may or may not apply to other forms of emails.

|Score | Name | Description | Advice |
|-------|--------|--------------|--------|
|-0.724 | MPART_ALT_DIFF | HTML and text parts are different | Make sure your TEXT version looks similar as the text inserted in your HTML version|
|-0.01 | T_REMOTE_IMAGE | Message contains an external image|

In addition to these, it lists "Your message does not contain a List-Unsubscribe header"

Is there anything we can do about this? At the very least, we can probably fix the text version, which looks to still be HTML. This might be causing most of the issue.

```
<html>
<p>
This is a test email to verify the SMTP configuration for <a href="<redacted>"><redacted></a>.
</p>
<p>When you can read this email it is probably configured correctly.</p>
</html>
```

**Comment by BlackDex** [comment]
It looks like you are not using the latest version of bitwarden_rs, or you have custom mail templates, because the message regarding similar text is fixed already. Also i did a test just a few days ago and i got a 10/10.

And i receive my mails also on Gmail, and the never ends up in the spam anymore.

**Reply by BlackDex** [reply]
@FractalBoy You could try the currently available `testing` tagged image on docker hub.
It has the latest changes regarding to mails.

The only other thing we could do is try to embed the images. But i'm not sure if we want that.
It could be a nice option for people to select or not.

**Reply by FractalBoy** [reply]
thanks, version 1.18 seemed to fix this.
